src/app/Dashboard/AutomatedAnalysis/ClickableAutomatedAnalysisLabel.tsx

```diff
@@ -0,0 +1,145 @@
+
+  const alertStyle = {
+    default: popoverStyles.modifiers.default,
+    info: popoverStyles.modifiers.info,
+    success: popoverStyles.modifiers.success,
+    warning: popoverStyles.modifiers.warning,
+    danger: popoverStyles.modifiers.danger,
+  };
+
+  const style = React.useMemo(() => {
+    if (isHoveredOrFocused) {
+      const defaultStyle = { cursor: 'pointer', '--pf-c-label__content--before--BorderWidth': '2.5px' };
+      if (isSelected) {
+        return { ...defaultStyle, '--pf-c-label__content--before--BorderColor': '#06c' };
+      }
+      return { ...defaultStyle, '--pf-c-label__content--before--BorderColor': '#8a8d90' };
+    }
+    return {};
+  }, [isSelected, isHoveredOrFocused]);
+
+  const colorScheme = React.useMemo((): LabelProps['color'] => {
+    // TODO: use label color schemes based on settings for accessibility
+    // context.settings.etc.
+    return label.score == -1
+      ? 'grey'
+      : label.score < AutomatedAnalysisScore.ORANGE_SCORE_THRESHOLD
+      ? 'green'
+      : label.score < AutomatedAnalysisScore.RED_SCORE_THRESHOLD
+      ? 'orange'
+      : 'red';
+  }, [label.score]);
+
+  const alertPopoverVariant = React.useMemo(() => {
+    return label.score == -1
+      ? 'default'
+      : label.score < AutomatedAnalysisScore.ORANGE_SCORE_THRESHOLD
+      ? 'success'
+      : label.score < AutomatedAnalysisScore.RED_SCORE_THRESHOLD
+      ? 'warning'
+      : 'danger';
+  }, [label.score]);
+
+  return (
+    <Popover
+      aria-label="automated-analysis-description-popover"
+      isVisible={isDescriptionVisible}
+      headerContent={<div className={`${clickableAutomatedAnalysisKey}-popover-header`}>{label.name}</div>}
+      alertSeverityVariant={alertPopoverVariant}
+      alertSeverityScreenReaderText={alertPopoverVariant}
+      shouldOpen={() => setIsDescriptionVisible(true)}
+      shouldClose={() => setIsDescriptionVisible(false)}
+      key={`${clickableAutomatedAnalysisKey}-popover-${label.name}`}
+      bodyContent={
+        <div
+          className={`${clickableAutomatedAnalysisKey}-popover-body`}
+          key={`${clickableAutomatedAnalysisKey}-popover-body-${label.name}`}
+        >
+          <p className={css(alertStyle[alertPopoverVariant], `${clickableAutomatedAnalysisKey}-popover-body-score`)}>
+            {label.score == -1 ? 'N/A' : label.score.toFixed(1)}
+          </p>
+          <p>{label.description}</p>
+        </div>
+      }
+    >
+      <Label
+        aria-label={`${label.name}`}
+        icon={<InfoCircleIcon />}
+        color={colorScheme}
+        style={style}
+        onMouseEnter={handleHoveredOrFocused}
+        onMouseLeave={handleNonHoveredOrFocused}
+        onFocus={handleHoveredOrFocused}
+        key={`${clickableAutomatedAnalysisKey}-${label.name}`}
+        isCompact
+      >
+        <span className={css(labelStyles.labelText)}>{`${label.name}`}</span>
```

Comment:
I think we can define our own css either as css selector or inline style to truncate texts as:

```typescript
{
  maxWidth: "16ch",
  whiteSpace: "nowrap",
  overflow: "hidden",
  textOverflow: "ellipsis"
}
```

References: https://css-tricks.com/snippets/css/truncate-string-with-ellipsis/

I guess this way we can avoid relying on that `--pf-c-label__text` style.

Reply:
Still we need to keep the comments to say that setting `isTruncate` will add a tooltip.